src/lazy/expanded/macro_table.rs

```diff
@@ -204,504 +202,195 @@ impl MacroTable {
-                        TemplateBodyExpr::element(
-                            TemplateBodyElement::with_value(TemplateValue::Symbol(Symbol::owned(
-                                "symbol_table",
-                            ))),
-                            ExprRange::new(2..3),
-                        ),
-                        // The module name $ion_encoding
-                        /* 3 */
-                        TemplateBodyExpr::element(
-                            TemplateBodyElement::with_value(TemplateValue::Symbol(Symbol::owned(
-                                "$ion_encoding",
-                            ))),
-                            ExprRange::new(3..4),
-                        ),
-                        // The `(macro_table ...)` s-expression.
-                        /* 4 */
-                        TemplateBodyExpr::element(
-                            TemplateBodyElement::with_value(TemplateValue::SExp),
-                            // Contains expression 4 (itself) through 8 (exclusive)
-                            ExprRange::new(4..8),
-                        ),
-                        // The clause name `macro_table`
-                        /* 5 */
-                        TemplateBodyExpr::element(
-                            TemplateBodyElement::with_value(TemplateValue::Symbol(Symbol::owned(
-                                "macro_table",
-                            ))),
-                            ExprRange::new(5..6),
-                        ),
-                        // The symbol literal `$ion_encoding`, indicating that we're appending
-                        // to the existing macro table.
-                        /* 6 */
-                        TemplateBodyExpr::element(
-                            TemplateBodyElement::with_value(TemplateValue::Symbol(Symbol::owned(
-                                "$ion_encoding",
-                            ))),
-                            ExprRange::new(6..7),
-                        ),
-                        // The variable at signature index 0 (`macro_definitions`)
-                        /* 7 */
-                        TemplateBodyExpr::variable(0, ExprRange::new(7..8)),
-                    ],
-                    annotations_storage: vec![Symbol::owned("$ion_encoding")],
-                },
-                expansion_analysis: ExpansionAnalysis::directive(),
-            })),
-            Arc::new(Macro::named(
+            ),
+            template(
+                r#"
+                        (macro set_symbols (symbols*)
+                             $ion_encoding::(
+                               // Set a new symbol table
+                               (symbol_table [(%symbols)])
+                               // Include the active encoding module macros
+                               (macro_table $ion_encoding)
+                             )
+                       )
+                    "#,
+            ),
```

Comment:
🪧 My next task is to switch over to the default module syntax, which means redefining these system template macros. That job will now be much easier--I can just revise this Ion 1.0 text.

Reply:
Wonderful, this is such a nice change.